### ISSUE TITLE
Docker releases, automatic as part of release workflow

### ISSUE THREAD

**Issue by astubbs** [3ll3d00d/ezbeq#79]
Would be great to get a docker image published as part of the release workflow.

- when a release tags are made, a github workflow will automatically publish a matching docker image to docker hub

There are a couple of docker image projects, but they're inactive / out of date:
https://github.com/jmery/ezbeq-docker
https://github.com/fattykidd/ezbeq-dc
https://hub.docker.com/search?q=ezbeq

**Comment by astubbs** [comment]
@3ll3d00d would you open to this, if I contribute the work? Do you already have a docker hub account? Would be great to get it published under your username to match this github repo.

**Comment by 3ll3d00d** [comment]
I use this for my setup

https://github.com/3ll3d00d/ezbeq-jr

Publishes to GitHub packages rather than dockerhub

Happy to extend it

**Comment by astubbs** [comment]
Investigating…
Qqs:
- Why no mention of this in the ezBEQ readme?
- What’s the -jr stand for?
- Does this build and publish an image for every ezBEQ release?

**Comment by 3ll3d00d** [comment]
> Why no mention of this in the ezBEQ readme?

because it's just what I use for my own setup (which I run on unraid)

> jr

jriver, i.e. doesn't need an additional binaries (unlike minidsp)

> Does this build and publish an image for every ezBEQ release?

yes see https://github.com/3ll3d00d/ezbeq/blob/main/.github/workflows/create-app.yaml#L108

**Comment by astubbs** [comment]
Got it.

In terms of implementation, would you be open to renaming it to -docker, and extending it to include the needed packages for minidsp? It seems to me that would be the least work, while only adding a small amount of extra complexity.

The other option would be to create a totally new repo, based of the other old ones, and add it to your build triggers for the main project.

**Comment by fattykidd** [comment]
> Got it.
> 
> In terms of implementation, would you be open to renaming it to -docker, and extending it to include the needed packages for minidsp? It seems to me that would be the least work, while only adding a small amount of extra complexity.
> 
> The other option would be to create a totally new repo, based of the other old ones, and add it to your build triggers for the main project.

Why not copy one of the other users docker files and build yourself an image that includes minidsp-rs ? 
I can see this becoming another headache for the maintainer when people just pull his image and then can't figure out how to add their usb devices(minidsp) to the container.

**Comment by astubbs** [comment]
> Why not...

I've already done [that](https://hub.docker.com/r/astubbs/ezbeq-docker). But, I don't want to, for the same reason why you wouldn't for software X. Why doesn't everyone build their own docker image for everything?

Context: I only use my MiniDSP over the offical wifi dongle.

If someone wants this to work with USB, they can supply a pull request :)

**Comment by astubbs** [comment]
Another option could be to add a trigger for https://github.com/jmery/ezbeq-docker to build and publish whenever ezbeq gets a release. However, IMO the docker project should ideally live right next to the official ezbeq repo, or even inside it. The author @jmery doesn't seem to be active and there's no exist trigger for his image to build on an ezbeq build. It's also no designed to pick up new versions of minidsp-rs.

**Comment by astubbs** [comment]
Ok, here's a first draft. Tested in my environment.

https://github.com/3ll3d00d/ezbeq-jr/pull/1

Let's discuss implementation specifics in the PR.

**Comment by astubbs** [comment]
> > Why no mention of this in the ezBEQ readme?
> 
> because it's just what I use for my own setup (which I run on unraid)

FAQ PR: https://github.com/3ll3d00d/ezbeq-jr/pull/2

**Comment by astubbs** [comment]
> Why not copy one of the other users docker files and build yourself an image that includes minidsp-rs ?

I understand your concern about managing USB access for minidsp devices in a Docker container. It’s true that Docker requires some additional configuration to pass through USB devices, such as using the --device flag or privileged mode. However, this is a one-time setup, and the centralised image simplifies everything else by including minidsp-rs pre-installed and ready to go.

Personally, I found this centralisation valuable. Just last week, I realized I was running an ezBEQ version that was about 10 months old because I had built my own image and forgotten about it. With a centralised image, users can stay up to date without worrying about manually rebuilding their environment or ensuring compatibility with the latest releases.

The benefits of a centralised Docker image include:
	1.	Pre-installed Support: Users don’t need to worry about manually adding or updating minidsp-rs.
	2.	Shared Improvements: Everyone benefits from updates and fixes contributed by the community.
	3.	Automated Updates: New builds are reliably triggered whenever ezBEQ releases a new version.
	4.	Environment Isolation: Docker ensures that the application runs in a consistent and compatible environment for all users.
	5.	Centralised Visibility and Trust: Hosting an official Docker build under the same GitHub organisation as ezBEQ ensures users can easily find and trust it. It provides a single, authoritative source for the project.

Additionally, by centralising our efforts, we can pool resources to keep the image working smoothly for everyone, rather than splitting off into separate, disconnected efforts. This benefits the community as a whole, ensuring a reliable and streamlined experience for all users.

For those concerned about USB passthrough, it’s worth noting that the same challenge applies whether you use a custom or centralised image. The centralised image, however, reduces complexity for everyone else, especially those using non-USB setups (like UDP).

Ultimately, the centralised image is designed to make things easier for most users while still allowing those with advanced needs to customise their own setup if necessary.
